pkg/api/types.go

```diff
@@ -2699,8 +2699,9 @@ type FinalizerName string
 
 // These are internal finalizer values to Kubernetes, must be qualified name unless defined here
 const (
-	FinalizerKubernetes FinalizerName = "kubernetes"
-	FinalizerOrphan     string        = "orphan"
+	FinalizerKubernetes       FinalizerName = "kubernetes"
+	FinalizerOrphanDependents string        = "orphan"
+	FinalizerDeleteDependents string        = "DeletingDependents"
```

Comment:
The others are lowercase, this should match them.

I would consider making this "foregroundDeletion" or "dependentDeletion".